File: plugins/extract-ts/testdata/cart/src/domain/basket/status.ts
export type BasketStatus = "open" | "checked-out" | "abandoned";

/** Where a basket can go from where it is: two ways out of open, none back. */
export const TRANSITIONS: Readonly<Record<BasketStatus, readonly BasketStatus[]>> = {
  open: ["checked-out", "abandoned"],
  "checked-out": [],
  abandoned: [],
};

export function canMove(from: BasketStatus, to: BasketStatus): boolean {
  return TRANSITIONS[from].includes(to);
}

File: plugins/extract-rust/testdata/oms/src/domain/order/status.rs
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum Status {
    Placed,
    Confirmed,
    Cancelled,
}

impl Status {
    pub fn as_str(&self) -> &'static str {
        match self {
            Status::Placed => "placed",
            Status::Confirmed => "confirmed",
            Status::Cancelled => "cancelled",
        }
    }
}

/// Where an order can go from where it is: two ways out of placed, none back.
pub const TRANSITIONS: &[(&str, &[&str])] = &[("placed", &["confirmed", "cancelled"]), ("confirmed", &[]), ("cancelled", &[])];
